tests/test_s3.py

```diff
@@ -164,6 +165,11 @@ def _check_presigned_url(self, url: str, expiration_time_seconds: int):
         # out to lunch for a while on a busy CI machine
         self.assertLessEqual(abs(expected_expiration_timestamp - actual_expiration_timestamp), 120)
 
+    def test_unthreaded_local(self):
+        S3ResourcePath.use_threads = False
+        self.test_local()
```

Comment:
If this test fails then `use_threads` will be stuck in `False`.

The test is tricky because you can't really test that anything changed inside boto, especially with moto mocking out much of boto.

Reply:
I'll go ahead and reset use_threads in the tearDown, which should fix that.